crates/cargo-gpu/src/install.rs

```diff
@@ -7,16 +7,70 @@ use crate::spirv_source::{
 use crate::{cache_dir, spirv_source::SpirvSource};
 use anyhow::Context as _;
 use cargo_metadata::Metadata;
-use log::{info, trace};
 use spirv_builder::SpirvBuilder;
 use std::path::{Path, PathBuf};
 
+/// Represents a functional backend installation, whether it was cached or just installed.
+#[derive(Clone, Debug)]
+#[expect(
+    clippy::exhaustive_structs,
+    reason = "never adding private members to this struct"
+)]
+pub struct InstalledBackend {
+    /// path to the `rustc_codegen_spirv` dylib
+    pub rustc_codegen_spirv_location: PathBuf,
+    /// toolchain channel name
+    pub toolchain_channel: String,
+    /// directory with target-specs json files
+    pub target_spec_dir: PathBuf,
+}
+
+impl InstalledBackend {
+    /// Creates a new `SpirvBuilder` configured to use this installed backend.
+    #[expect(
+        clippy::missing_panics_doc,
```

Comment:
Is `InstalledBackend` truly completely public? Or just within the crate? If it's just public within the crate then `pub (crate)` removes the `missing_panic_docs` lint warning.

Reply:
From the example above, `backend: InstalledBackend`.
```rust
    // install the toolchain and build the `rustc_codegen_spirv` codegen backend with it
    let backend = cargo_gpu::Install::from_shader_crate(shader_crate.clone()).run()?;
    [...]
    backend.to_spirv_builder(shader_crate, "spirv-unknown-vulkan1.2")
```

I like to give external crates access to the paths, in case they need to hack something. Though you could argue that we may want to `#[non_exhaustive]` that one, as in theory external crates shouldn't be touching it, but just call this function on it to create a new `SpirvBuilder`.